### PR TITLE
Deleting workflow execution info from all SQL tables

### PR DESCRIPTION
**Pull request by prathyushpv**
**What changed?**
Adding code to delete workflow info from all sql tables when the workflow is deleted.

**Why?**
Currently when a workflow gets deleted, only the data in executions table is deleted.
But there can be rows in following tables
ChildExecutionInfoMaps
ActivityInfoMaps
RequestCancelInfoMaps
SignalInfoMaps
TimerInfoMaps
SignalsRequestedSets
BufferedEvents

These rows will remain in these tables after the workflow is deleted.
Adding code to delete all corresponding rows in all these tables.

**How did you test it?**
Tested locally by creating workflows and checking if the table entries are deleted.

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: common/persistence/sql/execution.go [hunk 1]
```diff
@@ -601,13 +601,78 @@ func (m *sqlExecutionStore) DeleteWorkflowExecution(
...
-		RunID:       runID,
+	return m.txExecute(ctx, "DeleteWorkflowExecution", func(tx sqlplugin.Tx) error {
+		_, err := tx.DeleteAllFromChildExecutionInfoMaps(ctx, sqlplugin.ChildExecutionInfoMapsAllFilter{
+			ShardID:     request.ShardID,
+			NamespaceID: namespaceID,
+			WorkflowID:  request.WorkflowID,
+			RunID:       runID,
+		})
+		if err != nil {
+			return err
```
Comment (tdeebswihart) [inline common/persistence/sql/execution.go:612]: Can you annotate these with details on _which_ operation failed?

Comment (prathyushpv) [reply common/persistence/sql/execution.go]: Done.

**Review by yux0 — APPROVED** [review]

File: common/persistence/sql/execution.go [hunk 2]
```diff
@@ -601,13 +601,81 @@ func (m *sqlExecutionStore) DeleteWorkflowExecution(
...
-		RunID:       runID,
+	return m.txExecute(ctx, "DeleteWorkflowExecution", func(tx sqlplugin.Tx) error {
+		_, err := tx.DeleteAllFromChildExecutionInfoMaps(ctx, sqlplugin.ChildExecutionInfoMapsAllFilter{
+			ShardID:     request.ShardID,
+			NamespaceID: namespaceID,
+			WorkflowID:  request.WorkflowID,
+			RunID:       runID,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to execute DeleteAllFromChildExecutionInfoMaps: %w", err)
```
Comment (ast2023) [inline common/persistence/sql/execution.go:612]: Hi @prathyushpv, I would move this formatting into `tx.DeleteAll function` and at this level only adding that it was called from DeleteWorkflowExecution. This way log message will not only contain what function failed but also from where it was called from. But I am new to Go, may be Go does it automatically. @alexshtin what do you think?

Comment (prathyushpv) [reply common/persistence/sql/execution.go:612]: That makes sense, but these DeleteAll functions are just one line functions executing an sql qeury and returning result. Also, we are wrapping the error here, so we are not losing the original error.

**Pull request merged**